crates/re_time_panel/src/lib.rs

```diff
@@ -66,6 +66,10 @@ impl TimePanel {
         ui: &mut egui::Ui,
         time_panel_expanded: bool,
     ) {
+        // Naturally, many parts of the time panel need the time control.
+        // Copy it once, read/edit, and then write back at the end.
+        let mut time_ctrl = ctx.rec_cfg.time_ctrl.read().clone();
```

Comment:
I've tried very hard to figure out if this could lead to a race, e.g. some other place could write the current time that would the be overwritten (timeseries space view or play feature comes to mind), but I can't think of one really 🤷🏻‍♂️

Reply:
Yeah you're right, this is technically a very racy strategy 🤔

Reply:
Hmm I'm a bit torn now actually. If we have several parts of the viewer run in parallel and one of them is making any modifications to the time ctrl we would loose them.

Reply:
Too worried now that at some point someone does a secondary time affecting panel and gets ugly surprises. Pushed stuff to make it _better_, please have a look.

Reply:
Yeah, any kind of viewer-wide state behind a mutex is going to inevitably end up with nasty race conditions imo/ime.

We shouldn't ever modify anything beyond a system's local/private state while in the middle of a frame: we need a way to schedule these kinds of modifications so they run at the start of next frame imo.

Reply:
Yeah I fear your right, but I don't want to go through all of the necessary steps to go there. Also double buffering like we do on selection/hover now is a good alternative, that's still very well behaved in all situations IF you're ok with reading potentially outdated data at all callsites